game_event_player/tests.py

```diff
@@ -0,0 +1,81 @@
+from django.contrib.auth import get_user_model
+import pytest
+from django.db.utils import IntegrityError
+from django.core.exceptions import ValidationError
+from .models import GameEventPlayer, GameEvent
+from player.models import Player, BallGame
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def user():
+    return get_user_model().objects.create_user(
+        username='testuser',
+        password='testpass'
+    )
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def saved_player(user):
+    player = Player.objects.create(
+        user=user,
+        birth_date='1990-01-01',
+        favorite_ball_game=BallGame.Soccer
+    )
+    return player
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def saved_game_event():
+    game_event = GameEvent.objects.create()
+    return game_event
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def saved_game_event_player(saved_game_event, saved_player):
+    return GameEventPlayer.objects.create(game_event_id=saved_game_event, player_id=saved_player,
+                                          ball_responsible=False)
+
+
+@pytest.mark.django_db
+class TestGameEventPlayerModel:
+
+    def test_game_event_player_creation(self, saved_game_event, saved_player):
+        game_event_player = GameEventPlayer.objects.create(game_event_id=saved_game_event, player_id=saved_player,
+                                                           ball_responsible=False)
+        assert game_event_player.game_event_id == saved_game_event
+        assert game_event_player.player_id == saved_player
+        assert game_event_player.ball_responsible is False
+
+    def test_unique_together_constraint(self, saved_game_event, saved_player):
+        GameEventPlayer.objects.create(game_event_id=saved_game_event, player_id=saved_player, ball_responsible=False)
+        with pytest.raises(IntegrityError):
+            GameEventPlayer.objects.create(game_event_id=saved_game_event, player_id=saved_player,
+                                           ball_responsible=False)
+
+    def test_get_game_event_player(self, saved_game_event_player):
+        game_event_player = GameEventPlayer.objects.get(pk=saved_game_event_player.pk)
+        assert game_event_player == saved_game_event_player
+
+    def test_create_game_event_player_with_empty_player(self, saved_game_event):
+        with pytest.raises(IntegrityError):
+            GameEventPlayer.objects.create(game_event_id=saved_game_event, player_id=Player.objects.create(),
+                                           ball_responsible=False)
+
+    def test_create_game_event_player_with_empty_game_event(self, saved_player):
+        with pytest.raises(IntegrityError):
+            GameEventPlayer.objects.create(game_event_id=GameEvent.objects.create(), player_id=saved_player,
+                                           ball_responsible=False).full_clean()
```

Comment:
does `full_clean()` necessary here?
I think you should get an error even without it

Reply:
> does `full_clean()` necessary here? I think you should get an error even without it

You're right, the test passes without full_clean()